gui/src/app/pyodide/pyodideWorker/pyodideWorker.ts

```diff
@@ -0,0 +1,246 @@
+import { PyodideInterface, loadPyodide } from "pyodide";
+import {
+  MessageFromPyodideWorker,
+  isMessageToPyodideWorker,
+  PyodideWorkerMode,
+  PydodideWorkerStatus,
+} from "./pyodideWorkerTypes";
+
+let pyodideWorkerMode: PyodideWorkerMode | undefined = undefined;
+
+let pyodide: PyodideInterface | null = null;
+const loadPyodideInstance = async () => {
+  if (pyodide === null) {
+    if (!pyodideWorkerMode) {
+      throw Error("pyodideWorkerMode is not defined");
+    }
+    const packages =
+      pyodideWorkerMode === "data.py"
+        ? ["numpy", "micropip"]
+        : pyodideWorkerMode === "analysis.py"
+          ? ["numpy", "matplotlib", "pandas"]
+          : [];
```

Comment:
I think we can probably branch on the mode less often than we are. Might need to wait until implementing the data version to be sure